### ISSUE TITLE
Support Python 3.12

### ISSUE THREAD

**Issue by facelessuser** [facelessuser/Rummage#444]
wxPython Python 3.12 build is already available, but some dependencies, such as `babel` are not yet ready.

**Comment by facelessuser** [comment]
I was able to verify everything is working on Python 3.12. Dependencies have caught up.

**Comment by facelessuser** [comment]
I thought we could get this out, but it appears we have a segfault issue when closing Rummage using Py 3.12 and only Py 3.12. The app is completely functional, but we need to make sure this issue is not due to us doing or not doing some PY 3.12 specific thing that we need to change. If the issue is upstream in wxPython (which I suspect it is), then so be it; it is not our issue, but if not, we need a bit more time to track down whatever PY 3.12 quirk that exists that we now must workaround.

**Comment by facelessuser** [comment]
Running the same wxPython version in both Python 3.11 and Python 3.12, only Python 3.12 raises a segfault. This is also using the same dependency versions (at least at the top level).

If we were doing something fundamentally wrong, I would expect a segfault in both, but it is only Python 3.12. I imagine there is something incompatible, and probably unknown, in Python 3.12 causing this issue. Something may not be getting cleaned up etc., but I am struggling to figure out what it is.

If we call `wx.Exit()` we avoid the segfault, but we always return an error code of 255, that is even when we have absolutely no errors. `wx.Exit()` seems to always return 255.

We can call `os._exit()` which avoids general cleanup before exiting, and that also avoids the segfault, but is probably not ideal.

**Comment by facelessuser** [comment]
Okay, I am able to cause the failure with this simple example. It works on PY 3.11 and segfaults on exit for PY 3.12. There is absolutely nothing we are doing wrong. We'll hold off on a PY 3.12 release for now.

```py
import wx

class MyFrame(wx.Frame):
    def __init__(self, parent, title):
        wx.Frame.__init__(self, parent, title=title, size=(200,100))
        self.control = wx.TextCtrl(self, style=wx.TE_MULTILINE)
        self.Show(True)

def main():
    app = wx.App(False)
    frame = MyFrame(None, 'Small editor')
    app.MainLoop()

if __name__ == "__main__":
    main()
```

Everything is in place though and it could "technically" be used on PY 3.12, but it will segfault on exit.

**Comment by facelessuser** [comment]
We will check back on a future 3.12 RC, we may have to file an issue over on wxPython if this is not resolved before the official 3.12 release.

**Comment by facelessuser** [comment]
This issue is fixed in 4.2.2a1. We will await the official release of 4.2.2.

**Comment by facelessuser** [comment]
Python 3.12 is out, but we are still waiting for wxPython 4.2.2.

**Comment by facelessuser** [comment]
A new wxpython was released with wheels for Py 3.12 and Py 3.13. I belive this should have the fix for the issue in 3.12, and hopefully, all will be looking good for 3.13. I'll try to take a look at this soon to confirm.

**Comment by facelessuser** [comment]
Python 3.12 and 3.13 are working great with wxPython 4.2.2
